soda/bigquery/soda/data_sources/bigquery_data_source.py

```diff
@@ -153,6 +153,12 @@ def sql_get_table_names_with_count(
         where_clause = f"\nWHERE {table_filter_expression} \n" if table_filter_expression else ""
         return f"SELECT table_id, row_count \n" f"FROM {self.dataset_name}.__TABLES__" f"{where_clause}"
 
+    def sql_get_table(self, table_name: str, limit: int = None) -> str:
```

Comment:
@baturayo I think it's more appropriate to name it `sql_get_table_sample`

Reply:
Indeed makes sense👍

Reply:
This might sound wild (🤯 ) but why not go for `sql_select_star_with_limit`?

Reply:
interesting but more explanatory. Sample includes multiple different techniques (e.g random sampling, limit, stratified etc.) but limit is very specific. I can change it to `sql_select_star_with_limit`. Wdyt @tombaeyens ?

Reply:
`sql_select_star_with_limit` is an option but only a level lower.  At the top level, when entering the data_source, it should be per use-case.  So there should be 1 method that can be overridden for customizing only the samples fetching.

If there are multiple use cases that are based on the same query, those use case methods can delegate to a common `sql_select_star_with_limit` underneith.  But we should ensure that we don't have to refactor the data source class if a single data source impl wants to customize the query for a single use case like getting the samples.  It should be possible in the data source impl only by overriding the appropriate method.